packages/web/src/pages/upload-page/fields/availability/collectible-gated/CollectibleGatedFields.tsx

```diff
@@ -106,8 +107,11 @@ export const CollectibleGatedFields = ({
   // which makes the dropdown show the placeholder.
   // Otherwise, the default value is the nft collection which was previously selected,
   // which also includes the collection image.
-  const defaultCollectionName =
-    premiumConditionsValue?.nft_collection?.name ?? ''
+  const defaultCollectionName = isPremiumContentCollectibleGated(
```

Comment:
@dharit-tan You were correct on the bad merge. There are some new reads of the gating conditions on main that needed to be updated to use the union type.
Starting to wonder if this is actually useful as it requires calling multiple type checker functions each time we're trying to switch on the gate type? 😬

Reply:
yeah sigh.. it's still gives more type safety tho right?

Reply:
It does! And it's useful in the cases where you're checking for one specific condition before moving forward. I think it's clunky for the handful of cases where we are handling all of the types of premium content in one place.